### PR TITLE
Center element on paste

### PR DESCRIPTION
**Pull request by Fausto95**
Fixes https://github.com/excalidraw/excalidraw/issues/164

### CODE REVIEW

**Comment by Fausto95** [comment]
@dwelle this one should work

**Comment by dwelle** [comment]
Alright, this one works. Although, when you copy/paste several times, it pastes over the same position which may be hard to notice. Some older implem did offset it a bit, what do you think? (You'd still need to check that it doesn't move off-screen by the offset).

**Comment by Fausto95** [comment]
Well, I'm using Sketch right now and it has the same behavior, if we offset if, we might end being off-screen. So I think the current implementation is just fine!

File: src/index.tsx [hunk 1]
```diff
@@ -1009,17 +1009,11 @@ class App extends React.Component<{}, AppState> {
       parsedElements[0].type // need to implement a better check here...
     ) {
       clearSelection(elements);
-
-      if (x == null) x = 10 - this.state.scrollX;
-      if (y == null) y = 10 - this.state.scrollY;
-      const minX = Math.min(...parsedElements.map(element => element.x));
-      const minY = Math.min(...parsedElements.map(element => element.y));
```
Comment (vjeux) [inline src/index.tsx]: You can't remove this. It is useful for pasting multiple elements

Comment (Fausto95) [reply src/index.tsx]: 😬just realized it

**Comment by vjeux** [comment]
Doesn't seem to be working as expected, it pastes on the bottom-left'ish corner

image

**Comment by dwelle** [comment]
> Doesn't seem to be working as expected, it pastes on the bottom-left'ish corner

Bottom-*right'ish*, you mean?

@Fausto95 I'd simply compute the selection's bounding area (check [my algo in exportAsPNG](https://github.com/excalidraw/excalidraw/blob/master/src/scene/data.ts#L90), but use only selected elems), and paste them at the same location + offset, except when the bounding area would get off-screen, in which case decrease the offset so that it doesn't (there might be an edge case where you can't offset at all without screen-overflow, in which case I'd offset anyway -- the offset shouldn't be much more than +-10px, anyway).

**Comment by Fausto95** [comment]
> > Doesn't seem to be working as expected, it pastes on the bottom-left'ish corner
> 
> Bottom-_right'ish_, you mean?
> 
> @Fausto95 I'd simply compute the selection's bounding area (check [my algo in exportAsPNG](https://github.com/excalidraw/excalidraw/blob/master/src/scene/data.ts#L90), but use only selected elems), and paste them at the same location + offset, except when the bounding area would get off-screen, in which case decrease the offset so that it doesn't (there might be an edge case where you can't offset at all without screen-overflow, in which case I'd offset anyway -- the offset shouldn't be much more than +-10px, anyway).

It's centered correctly with your algo, but when it comes to pasting multiple elements, it completely messes up the position they were when copied

**Comment by Fausto95** [comment]
Now it is pasting under(well, right next) the mouse position

**Comment by vjeux** [comment]
It doesn't seem to be working as intended. It takes the delta between top left of screen and element, and reproduces this to cursor to element. So if you have an element outside of the top left, it's going to be pasted far away from the cursor.

What we want is to take the center of the element(s) we're pasting, and make this be at the position of the mouse.

ezgif-2-f007f044cb16

**Comment by vjeux** [comment]
Would be great to ship this relatively soon, the current behavior of moving the element at the top left is very annoying

**Comment by Fausto95** [comment]
It's working now

**Comment by vjeux** [comment]
Nice, just tested and it's working well!

**Pull request merged**